Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -8609,48 +8613,139 @@ def get_message_classes_dump(classes_as_list):
 
 def get_message_local_vars():
     """
-    A function that prints the local variables to log info
+    A function that returns the printable message about local variables
 
     :return: Message to print.
     :rtype: ``str``
     """
     local_vars = list(locals().items())
     ret_value = '\n\n--- Start Local Vars ---\n\n'
     for current_local_var in local_vars:
-        ret_value += str(current_local_var) + '\n'
+        ret_value += shorten_string_for_printing(str(current_local_var), 64) + '\n'
 
     ret_value += '\n--- End Local Vars ---\n\n'
 
     return ret_value
 
 
+def get_size_of_object(input_object):
+    """
+    A function that recursively iterate to sum size of object & members.
+
+    :type input_object: ``Any``
+    :param input_object: The object to calculate its memory footprint
+
+    :return: Size of input_object in bytes.
+    :rtype: ``int``
+    """
+    from collections import deque, Mapping
+    from numbers import Number
+    # Python 3 also needs range
+    # ZERO_DEPTH_BASES = (str, bytes, Number, range, bytearray)
+    ZERO_DEPTH_BASES = (str, bytes, Number, bytearray)
+    MAX_LEVEL = 20
+    _seen_ids = set()
+
+    def inner(obj, level):
+        if level == MAX_LEVEL:
+            # Stop at MAX_LEVEL
+            return sys.getsizeof(obj)
+        obj_id = id(obj)
+        if obj_id in _seen_ids:
+            return 0
+        _seen_ids.add(obj_id)
+        size = sys.getsizeof(obj)
+        if isinstance(obj, ZERO_DEPTH_BASES):
+            pass
+        elif isinstance(obj, (tuple, list, set, deque)):
+            size += sum(inner(i, level + 1) for i in obj)
+        elif isinstance(obj, Mapping):
+            mapping_items_keys = list(getattr(obj, 'keys')())
```

Comment:
why not just call `obj.keys()`?

Reply:
I'm not sure it's thread safe here. I got errors about the object changing on me at runtime.